apps/system/js/app_usage_metrics.js

```diff
@@ -613,19 +592,77 @@
     };
 
     var deviceInfoQuery = {
-      'deviceinfo.update_channel': 'unknown',
+      'app.update.channel': 'unknown',
+      'developer.menu.enabled': false, // If true, data is probably an outlier
+      'deviceinfo.hardware': 'unknown',
+      'deviceinfo.os': 'unknown',
       'deviceinfo.platform_version': 'unknown',
       'deviceinfo.platform_build_id': 'unknown',
-      'developer.menu.enabled': false // If true, data is probably an outlier
+      'deviceinfo.product_model': 'unknown',
+      'deviceinfo.software': 'unknown',
+      'deviceinfo.update_channel': 'unknown'
     };
 
     // Query the settings db to get some more device-specific information
-    AUM.getSettings(deviceInfoQuery, function(deviceinfo) {
-      data.deviceinfo = deviceinfo;
+    AUM.getSettings(deviceInfoQuery, function(deviceInfo) {
+      data.deviceinfo = deviceInfo;
+      data.simInfo = getSIMInfo();
+
       // Now transmit the data
       send(data);
     });
 
+    function getSIMInfo() {
```

Comment:
Yeah, that's a fair point. There are basically two buckets of data here: the ICC network (MCC/MNC/SPN registered on the SIM card), and the registered voice network (what the radio is physically connected to, could be roaming for example). Both of these data buckets could be provided as high level getters in SIMSlot, then formatted in this service..